packages/compass-query-history/src/stores/recent-list-store.js

```diff
@@ -25,7 +24,9 @@ const configureStore = (options = {}) => {
      * Filter attributes that aren't query fields or have default/empty values.
      * @param {object} attributes
      */
-    _filterDefaults(attributes) {
+    _filterDefaults(_attributes) {
+      // don't mutate the passed in parameter
```

Comment:
Lost some time here. My test that added the same query twice actually never executed the code that would check if it is a duplicate because this stripped ns from the query the first time, causing it to early return for a different reason.

I think it is just better style to not implicitly modify things that came in from the outside.